### PR TITLE
use Twitter::Extractor for creating links

### PR DESCRIPTION
**Pull request by masarakki**
I found a problem in creating link tag.

Conditions:

- Status contains mentions
- Status contains url with `=@username` (in mentions)

For example,

```
@masarakki

http://example.com/?x=@masarakki
```

In this case,
At first, `encode_and_link_urls` convert url to `<a href="http://example.com/?x=@masarakki" rel="nofollow ...">http://example.com/?x=@masarakki</a>`. 
Then, because of `=@username` **matches to Account::ACCOUNT_RE**, `@masarakki` will be convert to link by `link_mentions`. 
At last, it will be `<a href="http://example.com/?x=<a href="/...">@masarakki</a> rel="nofollow ...">`

Sample: https://friends.nico/@masarakki/2717988

### CODE REVIEW

**Comment by Gargron** [comment]
cc @abcang

**Comment by masarakki** [comment]
I think simple replacement is dangerous, it is better to use `placeholder` like replacement.

for example, 

html, links = encode_and_link_urls("@masakki Hello, world!,  http://example.com/?x=@masarakki")
# html: "@masarakki Hello, world!, \1", links: ['<a href="http://example.com/?x=@masarakki">http://example.com/?x=@masarakki</a>']
html, links = link_mentions(html, link)
# html: "\2 Hello, world!, \1", links: ['<a href="http://example.com/?x=@masarakki>...</a>' '<a href="/users/1">@masarakki</a>']
....

**Comment by abcang** [comment]
I think that it is safe to replace it after extracting the target like twitter-text.

File: app/lib/formatter.rb [hunk 1]
```diff
@@ -53,51 +50,66 @@ def encode(html)
...
+    end
+  end
 
-    chars = html.to_s.to_char_a
-    html_attrs = {
-      target: '_blank',
-      rel: 'nofollow noopener',
-    }
-    result = ''
+  def rewite(text, entities)
```
Comment (Gargron) [inline app/lib/formatter.rb]: Is this meant to be called rewrite?

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/lib/formatter.rb (hunk 1 above)
Comment (masarakki) [inline app/lib/formatter.rb]: oh sorry

**Review by Gargron — APPROVED** [review]

**Pull request merged**